util/library_loader.cc

```diff
@@ -0,0 +1,96 @@
+//  Copyright (c) 2011-present, Facebook, Inc.  All rights reserved.
+//  This source code is licensed under both the GPLv2 (found in the
+//  COPYING file in the root directory) and Apache 2.0 License
+//  (found in the LICENSE.Apache file in the root directory).
+
+#ifdef ROCKSDB_OPENSSL_AES_CTR
+#ifndef ROCKSDB_LITE
+
+#include "util/library_loader.h"
+
+#include <stdio.h>
+
+#include "rocksdb/status.h"
+
+namespace ROCKSDB_NAMESPACE {
+
+const char* UnixLibCrypto::crypto_lib_name_ = "crypto";
+
+LibraryLoader::LibraryLoader(const char* library_name) : is_valid_(false) {
+  Status stat = Env::Default()->LoadLibrary(library_name, std::string(), &lib_);
+  is_valid_ = stat.ok() && nullptr != lib_.get();
+}
+
+void* LibraryLoader::GetEntryPoint(const char* function_name) {
+  void* ret_ptr = {nullptr};
+
+  if (is_valid_) {
+    Status stat = lib_->LoadSymbol(function_name, &ret_ptr);
+    if (!stat.ok()) {
+      ret_ptr = nullptr;  // redundant but safe
+    }
+  }
+
+  return ret_ptr;
+}
+
+size_t LibraryLoader::GetEntryPoints(std::map<std::string, void*>& functions) {
+  size_t num_found{0};
+
+  if (is_valid_) {
+    for (auto& func : functions) {
+      void* tmp_ptr;
+
+      tmp_ptr = GetEntryPoint(func.first.c_str());
+      if (nullptr != tmp_ptr) {
+        ++num_found;
+        func.second = tmp_ptr;
+      }
+    }
+  }
+
+  return num_found;
+}
+
+UnixLibCrypto::UnixLibCrypto() : LibraryLoader(crypto_lib_name_) {
+  if (is_valid_) {
```

Comment:
Is there a reason to keep this in a separate file rather than in the env_openssl file?